src/sdk/namespace_s3.js

```diff
@@ -253,7 +253,8 @@ class NamespaceS3 {
         }
         dbg.log0('NamespaceS3.upload_object:', this.bucket, inspect(params), 'res', inspect(res));
         const etag = s3_utils.parse_etag(res.ETag);
-        return { etag, version_id: res.VersionId };
+        const last_modified_time = s3_utils.get_http_response_from_ret(res).httpResponse.headers.date;
```

Comment:
I suggest this will be it's own utility function `s3_utils.get_http_response_date(req)` that uses `s3_utils.get_http_response_from_ret()` inside, but it's best to hide these details from the namespace code if possible.